projects/ngrx.io/content/examples/store/src/app/state/allBooks.selectors.ts

```diff
@@ -0,0 +1,18 @@
+import { createSelector } from '@ngrx/store';
+import { AppState } from './state';
+import { Book } from '../book-list/books.service';
+
+export const selectBooks = createSelector(
+  (state: AppState) => state.books,
+  (books: Array<Book>) => books
+);
```

Comment:
Use a feature selector here instead

```suggestion
export const selectBooksState = createFeatureSelector<AppState, ReadonlyArray<Book>('books');
```